### ISSUE TITLE
ImportError: No module named lockfile

### ISSUE THREAD

**Issue by greenmoon55** [olipo186/Git-Auto-Deploy#104]
I've installed the package from ppa but I got the error.
I'm using Python 2.7.6 provided by Ubuntu 14.04.

> root@matlab4:~# service git-auto-deploy start
> Starting..
> Traceback (most recent call last):
>   File "/usr/bin/git-auto-deploy", line 9, in <module>
>     load_entry_point('git-auto-deploy==0.2.8', 'console_scripts', 'git-auto-deploy')()
>   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 351, in load_entry_point
>     return get_distribution(dist).load_entry_point(group, name)
>   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2363, in load_entry_point
>     return ep.load()
>   File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2088, in load
>     entry = __import__(self.module_name, globals(),globals(), ['__name__'])
>   File "/usr/lib/python2.7/dist-packages/gitautodeploy/__init__.py", line 2, in <module>
>     from lock import *
>   File "/usr/lib/python2.7/dist-packages/gitautodeploy/lock.py", line 1, in <module>
>     from lockfile import LockFile
> ImportError: No module named lockfile

**Comment by greenmoon55** [comment]
Still getting error `ImportError: cannot import name LockFile` after installing `python-lockfile`.

**Comment by rodolphopivetta** [comment]
Related with this, I got:

```
2016-06-10 18:20:02,491 [INFO ]  Exception happened during processing of request from
2016-06-10 18:20:02,492 [INFO ]  
2016-06-10 18:20:02,492 [INFO ]  ('10.0.2.2', 58469)
2016-06-10 18:20:02,492 [INFO ]  
2016-06-10 18:20:02,492 [ERROR]  Traceback (most recent call last):
2016-06-10 18:20:02,493 [ERROR]  File "/usr/lib/python2.7/SocketServer.py", line 284, in _handle_request_noblock
2016-06-10 18:20:02,494 [ERROR]  self.process_request(request, client_address)
2016-06-10 18:20:02,494 [ERROR]  File "/usr/lib/python2.7/SocketServer.py", line 310, in process_request
2016-06-10 18:20:02,494 [ERROR]  self.finish_request(request, client_address)
2016-06-10 18:20:02,494 [ERROR]  File "/usr/lib/python2.7/SocketServer.py", line 323, in finish_request
2016-06-10 18:20:02,495 [ERROR]  self.RequestHandlerClass(request, client_address, self)
2016-06-10 18:20:02,495 [ERROR]  File "/usr/lib/python2.7/SocketServer.py", line 638, in __init__
2016-06-10 18:20:02,495 [ERROR]  self.handle()
2016-06-10 18:20:02,496 [ERROR]  File "/usr/lib/python2.7/BaseHTTPServer.py", line 340, in handle
2016-06-10 18:20:02,496 [ERROR]  self.handle_one_request()
2016-06-10 18:20:02,497 [ERROR]  File "/usr/lib/python2.7/BaseHTTPServer.py", line 328, in handle_one_request
2016-06-10 18:20:02,497 [ERROR]  method()
2016-06-10 18:20:02,497 [ERROR]  File "gitautodeploy/httpserver.py", line 99, in do_POST
2016-06-10 18:20:02,498 [ERROR]  raise e
2016-06-10 18:20:02,498 [ERROR]  NameError: global name 'AlreadyLocked' is not defined
```

And in the code I see the AlreadyLocked class Exception isn't exists

**Comment by olipo186** [comment]
Hi,

There's a couple of problems here. When installing GAD from the debian package, one of the dependencies (python lockfile module) is not installed properly. I'll have to look into this and find a solution.

The debian distributed python-lockfile package seems to be outdated and does not expose the LockFile class. That's why you get the import errror even after installing the python-lockfile package.

For now, uninstall the python-lockfile debian package and install the latest lockfile module using pip instead;

    apt-get remove python-lockfile
    pip install lockfile

If you don't have pip installed, try installing it by running this from the command line:

    curl https://raw.github.com/pypa/pip/master/contrib/get-pip.py | python

Please let me know if this makes any difference.

**Comment by rodolphopivetta** [comment]
Anyway, the error in my previous comment exists apparently caused by the non-importation of the ```AlreadyLocked``` class, my suggestion is to add this in the first line of the [lock.py](https://github.com/olipo186/Git-Auto-Deploy/blob/master/gitautodeploy/lock.py#L1) file, like:

```python
from lockfile import LockFile, AlreadyLocked
```

**Comment by rodolphopivetta** [comment]
I created a pull request to solve this in https://github.com/olipo186/Git-Auto-Deploy/pull/105

**Comment by olipo186** [comment]
These issues should have been solved by PR #105 by @rodolphopivetta, commit fb67fb63bfe8abbd0e0ac0ab143c05be41d84c9f and commit c4407c007ab4ac900d6438761f72acae44878054. There's a new package (version 0.2.9) on the PPA that contains these fixes. Once it's built by launchpad (can take an hour or so) it should be available as usual;

    apt-get update
    apt-get upgrade git-auto-deploy

@greenmoon55, please reopen or create a new issue if the problem remains.

**Comment by greenmoon55** [comment]
It works now. Thanks a lot! @rodolphopivetta @olipo186
